Pinta.Effects/Effects/VoronoiDiagramEffect.cs

```diff
@@ -0,0 +1,284 @@
+	{
+		return distanceCalculationMethod switch {
+			DistanceCalculationMethod.Euclidean => Euclidean,
+			DistanceCalculationMethod.Manhattan => Manhattan,
+			DistanceCalculationMethod.Chebyshev => Chebyshev,
+			_ => throw new InvalidEnumArgumentException (
+				nameof (distanceCalculationMethod),
+				(int) distanceCalculationMethod,
+				typeof (DistanceCalculationMethod)),
+		};
+
+		static double Euclidean (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return difference.Magnitude ();
+		}
+
+		static double Manhattan (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return Math.Abs (difference.X) + Math.Abs (difference.Y);
+		}
+
+		static double Chebyshev (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return Math.Max (Math.Abs (difference.X), Math.Abs (difference.Y));
+		}
+	}
+
+	private static ImmutableHashSet<PointI> CreatePoints (RectangleI roi, int pointCount, RandomSeed pointLocationsSeed)
+	{
+		int effectivePointCount = Math.Min (pointCount, roi.Width * roi.Height);
+
+		Random randomPositioner = new (pointLocationsSeed.Value);
+		var result = ImmutableHashSet.CreateBuilder<PointI> (); // Ensures points' uniqueness
+
+		while (result.Count < effectivePointCount) {
+
+			PointI point = new (
+				X: randomPositioner.Next (roi.Left, roi.Right + 1),
+				Y: randomPositioner.Next (roi.Top, roi.Bottom + 1)
+			);
+
+			result.Add (point);
+		}
+
+		return result.ToImmutable ();
+	}
+
+	private static IEnumerable<ColorBgra> CreateColors (int colorCount, RandomSeed colorsSeed)
+	{
+		Random randomColorizer = new (colorsSeed.Value);
+		HashSet<ColorBgra> uniquenessTracker = new ();
+		while (uniquenessTracker.Count < colorCount) {
+			ColorBgra candidateColor = randomColorizer.RandomColorBgra ();
+			if (uniquenessTracker.Contains (candidateColor)) continue;
+			uniquenessTracker.Add (candidateColor);
+			yield return candidateColor;
+		}
+	}
+
+	public sealed class VoronoiDiagramData : EffectData
+	{
+		[Caption ("Distance Calculation Method")]
+		public DistanceCalculationMethod DistanceCalculationMethod { get; set; } = DistanceCalculationMethod.Euclidean;
+
+		[Caption ("Point Count"), MinimumValue (1), MaximumValue (1024)]
+		public int PointCount { get; set; } = 100;
+
+		// Translators: The user can choose whether or not to render the points used in the calculation of a Voronoi diagram
+		[Caption ("Show Points")]
+		public bool ShowPoints { get; set; } = false;
+
+		[Caption ("Color Sorting")]
+		public ColorSorting ColorSorting { get; set; } = ColorSorting.Random;
+
+		// Translators: In this context, "reverse" is a verb, and the user can choose whether or not they want to reverse the color sorting
+		[Caption ("Reverse Color Sorting")]
+		public bool ReverseColorSorting { get; set; } = false;
+
+		[Caption ("Colors Seed")]
```

Comment:
I'd maybe just call this `Random Colors` since the Seed is mentioned by the "Reseed" button underneath, and similar for the other seed below

Reply:
I went ahead and called that `Random Colors`, and similarly for the points `Random Point Locations`